analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ObjectsShouldNotBeDisposedMoreThanOnce.vb

```diff
@@ -0,0 +1,242 @@
+    End Sub
+
+    Public Sub Close_ParametersOfSameType(ByVal withDispose1 As IWithDispose, ByVal withDispose2 As IWithDispose)
+        withDispose1.Dispose()
+        withDispose2.Dispose()
+    End Sub
+
+    Public Sub Close_OneParameterDisposedThrice(ByVal withDispose1 As IWithDispose, ByVal withDispose2 As IWithDispose)
+        withDispose1.Dispose()
+        withDispose1.Dispose()  ' Noncompliant
+        withDispose1.Dispose()  ' Noncompliant
+        withDispose2.Dispose()
+    End Sub
+
+End Class
+
+Public Class DisposableAsync
+    Implements IDisposable, IAsyncDisposable
+
+    Public Sub Dispose() Implements IDisposable.Dispose
+    End Sub
+
+    Public Function DisposeAsync() As ValueTask Implements IAsyncDisposable.DisposeAsync
+    End Function
+
+End Class
+
+Public Class DisposeAsync
+
+    Private Async Function DisposeAsyncTwiceUsingStatement(ByVal d As DisposableAsync) As Task
+        Using d ' FN
+        End Using
+        Await d.DisposeAsync()
+    End Function
+
+    Private Async Function DisposeAsyncTwice() As Task
+        Dim d = New DisposableAsync()
+        Await d.DisposeAsync()
+        Await d.DisposeAsync() ' Noncompliant
+    End Function
+
+    Private Async Function DisposeTwiceMixed() As Task
+        Dim d = New DisposableAsync()
+        Await d.DisposeAsync()
+        d.Dispose()  ' Noncompliant
+    End Function
+
+End Class
+
+Public Class Class1
+    Implements IDisposable
+
+    Private Sub Dispose() Implements IDisposable.Dispose
+    End Sub
+
+    Public Sub DisposeMultipleTimes()
+        Dispose()
+        Me.Dispose() ' FN
+        Dispose() ' FN
+    End Sub
+
+    Public Sub DoSomething()
+        Dispose()
+    End Sub
+
+End Class
+
+Class TestLoops
+
+    Public Shared Sub LoopWithBreak(ByVal list As String(), ByVal condition As Boolean, ByVal withDispose As IWithDispose)
+        For Each x As String In list
+            Try
+                If condition Then
+                    withDispose.Dispose() ' FN
+                End If
+
+                Exit For
+            Catch __unusedException1__ As Exception
+                Continue For
+            End Try
+        Next
+    End Sub
+
+    Public Shared Sub LoopMethod(ByVal list As String(), ByVal condition As Boolean, ByVal withDispose As IWithDispose)
+        For Each x As String In list
+            If condition Then
+                withDispose.Dispose()  ' Noncompliant
+            End If
+        Next
+    End Sub
+End Class
+
+Class UsingDeclaration
+
+    Public Sub Disposed_UsingStatement()
+        Using d = New Disposable()  ' Noncompliant
+            d.Dispose()
+        End Using
+    End Sub
+
+End Class
+
+Public Class Close
+
+    Public Sub CloseStreamTwice()
+        Dim fs = New FileStream("c:\foo.txt", FileMode.Open)
+        fs.Close()
+        fs.Close() ' FN - Close On streams is disposing resources
+    End Sub
+
+    Private Sub CloseTwiceDBConnection(ByVal connection As DbConnection)
```

Comment:
```suggestion
    Private Sub CloseTwiceDBConnection(ByVal Connection As DbConnection)
```

Reply:
Test removed after this comment https://github.com/SonarSource/sonar-dotnet/pull/7414#discussion_r1229744448